content/posts/pyodide-react-pandas-matplotlib-scikit.md

```diff
@@ -0,0 +1,224 @@
+---
+title: "Visual Low/No Code Data Preparation and Analysis Web App Built with Pyodide and React"
+date: 2024-06-19T01:39:18-04:00
+draft: true
+tags: ["announcement", "guest post"]
+author: "Raul Andrial"
+# author: ["Me", "You"] # multiple authors
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+description: null
+# canonicalURL: "https://canonical.url/to/page"
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+cover:
+    # image: "<image path/url>" # image path/url
+    # alt: "<alt text>" # alt text
+    # caption: "<text>" # display caption under cover
+    relative: false # when using page bundles set this to true
+    hidden: true # only hide on current single page
+---
+
+Hello, my name is [Raul Andrial](https://randr000.github.io/portfolio-resume/tech) and I am a software engineer located in Miami, FL, USA. I created a proof of concept web app with Pyodide that allows users to use the pandas library without needing to code but can use code if they want to. The app uses a drag and drop interface. I also included functionality for plotting with matplotlib and linear regression using scikit-learn.
+
+You can read more about it [here](https://github.com/randr000/react_Pyodide_data_prep).
+
+You can test it out [here](https://randr000.github.io/react_Pyodide_data_prep/).
+
+## Pyodide Context Provider
+
+The Pyodide instance is loaded and stored in its own context using React's [Context API](https://react.dev/reference/react/createContext). In order for any component to run Python code it must be a child of the PyodideContextProvider component.
+
+```
+import React, { useEffect, useRef, useState } from 'react';
+import { loadPyodide } from 'Pyodide/Pyodide.js';
+
+export const PyodideContext = React.createContext(null);
+
+export const PyodideContextProvider = ({children, toLoadPyodide=true}) => {
+
+    const [isPyodideLoaded, setIsPyodideLoaded] = useState(!toLoadPyodide);
+    const PyodideRef = useRef(null);
+
+    useEffect(() => {
+        toLoadPyodide &&
+        (async function () {
+            PyodideRef.current = await loadPyodide({
+                indexURL : "https://cdn.jsdelivr.net/Pyodide/v0.21.3/full/"
+            });
+
+            // Load Python packages
+            ['pandas', 'numpy', 'matplotlib', 'scikit-learn', 'cloudpickle'].forEach(async pkg => await PyodideRef.current.loadPackage(pkg));
```

Comment:
you don't need to pass packages one-by-one. `pyodide.loadPackage('pandas', 'numpy', 'matplotlib', 'scikit-learn', 'cloudpickle']`) will do the job.

Reply:
Changed.